config/routes.rb

```diff
@@ -1,11 +1,15 @@
 Rails.application.routes.draw do
   devise_for :users
+  get 'recipes/index', to: 'recipes#public'
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
   # root "articles#index"
-  resources :users
+
+  root "recipes#public"
+  resources :users do
+    resources :recipes
+  end
   resources :foods
   resources :recipes
```

Comment:
Here I believe we only need the routes to recipes to be defined once. Why do you think recipes routes should be nested inside users?